engines/e_padlock.c

```diff
@@ -649,6 +970,212 @@ padlock_aes_init_key(EVP_CIPHER_CTX *ctx, const unsigned char *key,
     return 1;
 }
 
+/* Prepare the encryption key for GMI sm4  usage */
+static int
+gmi_sm4_init_key(EVP_CIPHER_CTX *ctx, const unsigned char *key,
+                 const unsigned char *iv, int enc)
+{
+    struct gmi_cipher_data *cdata;
+    unsigned long mode = EVP_CIPHER_CTX_mode(ctx);
+ 
+    if (key == NULL)
+        return 0;               /* ERROR */
+
+    cdata = ALIGNED_CIPHER_DATA_GMI(ctx);
+    memset(cdata, 0, sizeof(*cdata));
+
+    /* Prepare Control word. */
+    if (mode == EVP_CIPH_OFB_MODE || mode == EVP_CIPH_CTR_MODE)
+        cdata->cword.b.encdec = 0;
+    else
+        cdata->cword.b.encdec = (EVP_CIPHER_CTX_encrypting(ctx) == 0);
+
+    cdata->cword.b.func = CCS_ENCRYPT_FUNC_SM4;
+    cdata->cword.b.mode = 1<<(mode-1);;
+    cdata->cword.b.digest = 0;
+
+    if(iv != NULL)
+    {
```

Comment:
curly bracket should be placed in the same line of `if` statement. Besides, there should be a space between `if` and `(`.

Reply:
Ahh, this `if` block has only one statement, so the curly brackets are not necessary.